### DISCUSSION TITLE
Merge with Alligator?

### DISCUSSION THREAD

**Discussion by plata** [cscarney/syndic#54] [6: Development]
Syndic seems to be very similar to [Alligator](https://invent.kde.org/plasma-mobile/alligator). Wouldn't it make sense to join forces there?

**Comment by cscarney** [comment]
Back when I started this project in 2020, Alligator was specifically focused on Plasma Mobile, and I was much more interested in making a casual feed browser for the desktop. It does look like Alligator is a bit more functional on the desktop these days, and I've created an Android version of Syndic, so there's probably more overlap now than there was then.

I'm not sure how a "merge" would work, though, since Syndic's code base is quite different from theirs.  Syndic uses a lot of the same components (QML, Kirigami, Sqlite, Syndication), but put together differently in ways that would make it difficult to directly merge its features into Alligator.

I'm open to the possibility of contributing to Alligator, though, and I'll be on the lookout for bug reports that look like issues I've already solved.

**Reply by plata** [reply]
I didn't compare the apps in detail but I had the impression that they provide pretty much the same functionality. If that's really the case, it means that any other feature would have to be implemented twice (which would be a waste of man power, really).

When I wrote "merge", I thought about it on a functional level (not a code merge).

The main thing I'm missing in either app is: https://invent.kde.org/plasma-mobile/alligator/-/merge_requests/32

**Reply by cscarney** [reply]
> I didn't compare the apps in detail but I had the impression that they provide pretty much the same functionality. If that's really the case, it means that any other feature would have to be implemented twice (which would be a waste of man power, really).
>
> When I wrote "merge", I thought about it on a functional level (not a code merge).

Would it be a better use of my time to work on Alligator instead of maintaining Syndic?  Maybe.  But I *use* Syndic every day as my primary feed reader app, while I've never actually managed to get Alligator working acceptably with a non-trivial collection of feeds.  So I'm probably going to keep working on the product I actually use :)

They do have a lot of the same functionality, but a lot of that just comes from being in the same product category.  There are certain things that every feed reader app has to do to be functional.  I've been trying to target a niche somewhere in-between Alligator (mobile-first) and Akregator (desktop only) and create a responsive design that feels at home on the desktop while still being touch-friendly.

> The main thing I'm missing in either app is: https://invent.kde.org/plasma-mobile/alligator/-/merge_requests/32

Scraping web content is something I've been thinking about for a while, and putting off doing.  The problem is that I don't want to depend on nodejs, and "call an outside process if it's installed" doesn't work well with the flatpak sandbox.  Which probably means trying to re-implement Readability's algorithm in c++, which is a giant PITA.

That said, your Alligator patch has a nice clean abstraction layer, so it could probably be used as a starting point that would work for non-sandboxed installs, and plug an internal implementation into it later.  If I use the same API then any future work on the implementation could be contributed back to Alligator.  I might look into that if I get some free time this week.

**Reply by cscarney** [reply]
Made a quick proof-of-concept using your readability code from Alligator: #55
